jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/PluginConfigurationProcessor.java

```diff
@@ -230,4 +259,47 @@ MavenSettingsServerCredentials getMavenSettingsServerCredentials() {
   boolean isBaseImageCredentialPresent() {
     return isBaseImageCredentialPresent;
   }
+
+  /**
+   * Compute the container entrypoint, in this order :
+   *
+   * <ol>
+   *   <li>the user specified one, if set
+   *   <li>for a war project, the jetty default one
+   *   <li>for a jar project, by resolving the main class
+   * </ol>
+   *
+   * @param logger the logger used to display messages.
+   * @param jibPluginConfiguration the {@link JibPluginConfiguration} providing the configuration
+   *     data
+   * @param projectProperties used for providing additional information
+   * @return the entrypoint
+   * @throws MojoExecutionException if the http timeout system property is misconfigured
+   */
+  static List<String> computeEntrypoint(
+      Log logger,
+      JibPluginConfiguration jibPluginConfiguration,
+      MavenProjectProperties projectProperties)
+      throws MojoExecutionException {
+
+    List<String> entrypoint = jibPluginConfiguration.getEntrypoint();
+    if (!entrypoint.isEmpty()) {
+      if (jibPluginConfiguration.getMainClass() != null
+          || !jibPluginConfiguration.getJvmFlags().isEmpty()) {
+        logger.warn("<mainClass> and <jvmFlags> are ignored when <entrypoint> is specified");
+      }
+    } else {
+      if (isWarPackaging(jibPluginConfiguration)) {
```

Comment:
There's `MavenProjectProperties.isWarProject()`, much like `GradleProjectProperties.isWarProject()`.

Although honestly I'm not sure if having `ProjectProperties.isWarProject()` is incredibly useful and it may make sense to the method from the interface. We can think about this, along with Gradle.

Reply:
I don't have MavenProjectProperties everywhere.

Reply:
I see `MavenProjectProperties` being passed into the method as an argument.

Reply:
Never mind. I see you need the `isWarPackaging` method in this class anyway.